src/Application.vala

```diff
@@ -25,6 +25,8 @@ public class Mail.Application : Gtk.Application {
     }
 
     construct {
+        Environment.set_variable ("GTK_USE_PORTAL", "1", true);
```

Comment:
i think we shouldn't try to override the environment if already set externally. also a FIXME or TODO comment saying "remove this once we have Gtk.FileDialog" would be good, so that we don't forget to remove this workaround later.

Reply:
Ah does Gtk.FileDialog always use portals even on the host?

We're currently doing this in Code and System Settings for the same reason: https://github.com/search?q=org%3Aelementary+Environment.set_variable+%28%22GTK_USE_PORTAL%22%2C+%221%22%2C+true%29%3B&type=code

Reply:
yes, they try the portal first and if that fails, they use a fallback Gtk.FileChooser.